packages/jest-resolve/src/shouldLoadAsEsm.ts

```diff
@@ -68,11 +71,29 @@ function shouldLoadAsEsm(path: Config.Path): boolean {
 
 function cachedPkgCheck(cwd: Config.Path): boolean {
   // TODO: can we cache lookups somehow?
```

Comment:
can delete this comment now, the caching added here is as far as we can feasibly go, I think. I was thinking some sort of handling of parent-child directories, but I don't think it's worth it